From 2006a0addc288e87aa03bf3835bafd786aadefa0 Mon Sep 17 00:00:00 2001
From: James Ross-Gowan <rossy@jrg.systems>
Date: Thu, 25 Jan 2018 22:59:49 +1100
Subject: [PATCH] utils: avoid ffsll()

ffsll() is available in glibc and BSD, but it's non portable and
unavailable on Windows. Use __builtin_ffsll() instead. This function
already uses GCC builtins anyway.
---
 src/utils/upload.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utils/upload.c b/src/utils/upload.c
index ecc03af..df668aa 100644
--- a/src/utils/upload.c
+++ b/src/utils/upload.c
@@ -47,7 +47,7 @@ void pl_plane_data_from_mask(struct pl_plane_data *data, uint64_t mask[4])
 
     for (int i = 0; i < PL_ARRAY_SIZE(comps); i++) {
         comps[i].size = __builtin_popcount(mask[i]);
-        comps[i].shift = PL_MAX(0, ffsll(mask[i]) - 1);
+        comps[i].shift = PL_MAX(0, __builtin_ffsll(mask[i]) - 1);
     }
 
     // Sort the components by shift